src/sqlfluff/core/parser/parsers.py

```diff
@@ -43,21 +41,28 @@ def is_optional(self) -> bool:
     def _is_first_match(self, segment: BaseSegment):
         """Does the segment provided match according to the current rules."""
 
-    def _make_match_from_first_result(self, segments: Tuple[BaseSegment, ...]):
+    def _make_match_from_first_result(self, segment: BaseSegment):
```

Comment:
Nitpick: Since the function now takes a single segment rather than a tuple of segments, we may want to rename it (the `first_result` bit no longer makes sense).